network/p2p/p2p_setup.go

```diff
@@ -115,6 +117,52 @@ func (n *p2pNetwork) IsBadPeer(logger *zap.Logger, peerID peer.ID) bool {
 	return n.idx.IsBad(logger, peerID)
 }
 
+func (n *p2pNetwork) canAcceptNewConnection(lg *zap.Logger, peerID peer.ID) bool {
+	if n.idx == nil {
+		return false
+	}
+
+	maxPeers := n.cfg.MaxPeers
+	inBoundLimit := int(float64(n.cfg.MaxPeers) * inboundRatio)
+
+	if maxPeers < inBoundLimit {
+		return false
+	}
```

Comment:
Not sure what this condition checks for, isn't it always wrong (`inBoundLimit` should always be >=  `maxPeers`) ?